src/components/ArrayInput/index.js

```diff
@@ -0,0 +1,501 @@
+import utils from '../../helpers/utils';
+import './style.scss';
+
+import Element from '../Element';
+import Container from '../Container';
+import Panel from '../Panel';
+import NumericInput from '../NumericInput';
+import Button from '../Button';
+
+const CLASS_ARRAY_INPUT = 'pcui-array-input';
+const CLASS_ARRAY_EMPTY = 'pcui-array-empty';
+const CLASS_ARRAY_SIZE = CLASS_ARRAY_INPUT + '-size';
+const CLASS_ARRAY_CONTAINER = CLASS_ARRAY_INPUT + '-items';
+const CLASS_ARRAY_ELEMENT = CLASS_ARRAY_INPUT + '-item';
+const CLASS_ARRAY_DELETE = CLASS_ARRAY_ELEMENT + '-delete';
+
+var DEFAULTS = {
```

Comment:
How are we going to support additional types is there a way to register new DEFAULTS for new types?

Reply:
I've had to change the way the array input class receives it's input element as there's no global element registration in pcui. Now the input class you want to use is passed in as an argument, along with the type string. We'll need to add in defaults here as new types are added. But it also means users wont be importing every input element class into their project every time they use the array input class.